### PR TITLE
fix: make rln rate limit spec compliant

### PR DESCRIPTION
**Pull request by alrevuelta**
fix: make rln rate limit spec compliant
closes https://github.com/waku-org/nwaku/issues/2289

### CODE REVIEW

**Review by fryorcraken — APPROVED** [review]

**Review by jm-clius — APPROVED** [review]

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim [hunk 1]
```diff
@@ -445,7 +445,7 @@ procSuite "WakuNode - RLN relay":
       proofAdded1 = node1.wakuRlnRelay.appendRLNProof(wm1, time)
       # another message in the same epoch as wm1, it will break the messaging rate limit
       wm2 = WakuMessage(payload: "message 2".toBytes(), contentTopic: contentTopic)
-      proofAdded2 = node1.wakuRlnRelay.appendRLNProof(wm2, time + EpochUnitSeconds)
+      proofAdded2 = node1.wakuRlnRelay.appendRLNProof(wm2, time)
```
Comment (alrevuelta) [inline tests/waku_rln_relay/test_wakunode_rln_relay.nim:448]: @rymnc Unsure I understand why `wm2` had `time + EpochUnitSecods` if its expected to "break the messaging rate limit". Isn't it better to set the same time?

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim [hunk 2]
```diff
@@ -469,25 +469,25 @@ procSuite "WakuNode - RLN relay":
...
           completionFut3.complete(true)
-    
+
     # mount the relay handler for node2
     node2.subscribe((kind: PubsubSub, topic: DefaultPubsubTopic), some(relayHandler))
     await sleepAsync(2000.millis)
 
     await node1.publish(some(DefaultPubsubTopic), wm1)
-    await sleepAsync(10.seconds)
     await node1.publish(some(DefaultPubsubTopic), wm2)
```
Comment (alrevuelta) [inline tests/waku_rln_relay/test_wakunode_rln_relay.nim:478]: @rymnc Mind clarifying why this sleep was needed? Epoch is hardcoded above, so perhaps not needed?

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim [hunk 3]
```diff
@@ -469,25 +469,25 @@ procSuite "WakuNode - RLN relay":
...
       res2 = await completionFut2.withTimeout(10.seconds)
       res3 = await completionFut3.withTimeout(10.seconds)
 
     check:
-      (res1 and res2 and res3) == true # all 3 are valid
-      node2.wakuRlnRelay.nullifierLog.len() == 1 # after clearing, only 1 is stored
+      res1 == true
+      res2 == false
+      res3 == true
+      node2.wakuRlnRelay.nullifierLog.len() == 2
```
Comment (alrevuelta) [inline tests/waku_rln_relay/test_wakunode_rln_relay.nim:490]: @rymnc The change from `10` to `1` in `EpochUnitSeconds` triggered this edge case (I belive). Mind clarifying what would be expected here?

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim [hunk 4]
```diff
@@ -469,25 +469,25 @@ procSuite "WakuNode - RLN relay":
...
     let
       res1 = await completionFut1.withTimeout(10.seconds)
       res2 = await completionFut2.withTimeout(10.seconds)
       res3 = await completionFut3.withTimeout(10.seconds)
 
     check:
-      (res1 and res2 and res3) == true # all 3 are valid
-      node2.wakuRlnRelay.nullifierLog.len() == 1 # after clearing, only 1 is stored
+      res1 == true
+      res2 == false
```
Comment (alrevuelta) [inline tests/waku_rln_relay/test_wakunode_rln_relay.nim:488]: @rymnc Not sure why `rest2` was expected to complete. Since as stated above the second message "it will break the messaging rate limit" and that implies that the handler wont be triggered? So `res==false` would be expected isn't it?

**Review by alrevuelta — COMMENTED** [review]
@rymnc This change triggered some edge cases? Mind checking my questions?

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim (hunk 1 above)
Comment (rymnc) [reply tests/waku_rln_relay/test_wakunode_rln_relay.nim:448]: doesn't matter too much, just showing that 2 messages were generated within the same epoch window.

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim (hunk 2 above)
Comment (rymnc) [reply tests/waku_rln_relay/test_wakunode_rln_relay.nim:478]: it was for message propagation i believe, due to some race condition we had earlier

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim (hunk 4 above)
Comment (rymnc) [reply tests/waku_rln_relay/test_wakunode_rln_relay.nim:488]: agree, oversight on my part

File: tests/waku_rln_relay/test_wakunode_rln_relay.nim (hunk 3 above)
Comment (rymnc) [reply tests/waku_rln_relay/test_wakunode_rln_relay.nim:490]: this is correct, thanks for the fix 😁

**Review by rymnc — APPROVED** [review]
LGTM, thanks for the fixes in the edge cases!
